### PR TITLE
Enable fake creation via constructors with in parameters

### LINKED ISSUE #1948: Failure to create fake via constructor with 'in' parameter
FakeItEasy will not attempt to create a fake via any constructor with an `in` parameter. Instead, it will give an error indicating that the `in` parameter Type "could not be resolved".

Here is an example test that does not work.

```csharp
public class ClassWithConstructorInParameter
{
	public ClassWithConstructorInParameter(in DateTimeOffset dateTimeOffset) { }
}

[TestMethod]
public void CreateFakeViaConstructorWithInParameter()
{
	A.Fake<ClassWithConstructorInParameter>();
}
```

I will submit a pull request shortly with a suggested fix.

### PR DESCRIPTION
**Pull request by mriehm**
Fixes #1948

### CODE REVIEW

File: tests/FakeItEasy.Specs/CreationSpecs.cs [hunk 1]
```diff
@@ -425,6 +425,19 @@ public void ClassWithPrivateConstructorCannotBeFaked(Exception exception)
...
+        {
+            "Given a class with a constructor with an in parameter"
+                .See<ClassWithConstructorInParameter>();
+
+            "When I create a fake of the class"
+                .x(() => fake = this.CreateFake<ClassWithConstructorInParameter>());
+
+            "Then the fake is created"
+                .x(() => fake.Should().BeAFake());
+        }
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Specs/CreationSpecs.cs:439]: Could you please add tests with `ref` and `out` parameters as well? Just for completeness.

**Review by thomaslevesque — COMMENTED** [review]
Thanks for the PR @mriehm.
I just have one request to add coverage for `ref` and `out` parameters as well. I think your fix works for those too, but we should make sure.

**Review by thomaslevesque — APPROVED** [review]
Thanks for the update, @mriehm! Looks good to me.
@blairconrad, any objection to merging this?

**Review by blairconrad — APPROVED** [review]

**Comment by blairconrad** [comment]
Thanks, @mriehm. Nice addition.

**Pull request merged**